From 558609c439a363f25124073bc02aa42ab96c33b5 Mon Sep 17 00:00:00 2001
From: Harini Katakam <harini.katakam@xilinx.com>
Date: Mon, 6 Jul 2020 13:56:34 +0530
Subject: [PATCH 1418/1852] net: gmii2rgmii: Fix warning for return variable
 check

commit 19b191605a4034754f04e40a62a9a25b4c085266 from
https://github.com/Xilinx/linux-xlnx.git

Check return of mdiobus_write for errors.

Addresses-coverity: ("check_return")
Signed-off-by: Harini Katakam <harini.katakam@xilinx.com>
Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/net/phy/xilinx_gmii2rgmii.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/net/phy/xilinx_gmii2rgmii.c b/drivers/net/phy/xilinx_gmii2rgmii.c
index 2d1449345959..d4f10fa4ecb2 100644
--- a/drivers/net/phy/xilinx_gmii2rgmii.c
+++ b/drivers/net/phy/xilinx_gmii2rgmii.c
@@ -52,9 +52,7 @@ static int xgmiitorgmii_read_status(struct phy_device *phydev)
 	else
 		val |= BMCR_SPEED10;
 
-	mdiobus_write(bus, addr, XILINX_GMII2RGMII_REG, val);
-
-	return 0;
+	return mdiobus_write(bus, addr, XILINX_GMII2RGMII_REG, val);
 }
 
 static int xgmiitorgmii_probe(struct mdio_device *mdiodev)
-- 
2.31.1

